UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -1083,5 +1120,23 @@ public static void ObservationToTexture(Camera obsCamera, int width, int height,
             RenderTexture.active = prevActiveRT;
             RenderTexture.ReleaseTemporary(tempRT);
         }
-    }
+
+        /// <summary>
+        /// Sets the custom observation for the agent for this episode.
+        /// </summary>
+        /// <param name="customObservation">New value of the agent's custom observation.</param>
+        public void SetCustomObservation(CustomObservation customObservation)
+        {
+            info.customObservation = customObservation;
+        }
+
+        /// <summary>
+        /// Get the custom observation that has previously been set for this agent during this episode.
+        /// </summary>
+        /// <returns> An instance of a CustomObservation message.</returns>
+        public CustomObservation GetCustomObservation()
```

Comment:
I am not sure this method is useful at all. I would remove it.

Reply:
Done